commonutils/src/main/java/io/intrepid/commonutils/AndroidStringUtils.java

```diff
@@ -89,4 +96,49 @@ public static Spanned fromHtml(String source, int flags) {
             return Html.fromHtml(source);
         }
     }
+
+    /**
+     * Attempts to parse the given String to a Uri.
+     * This method exists because standard pattern matching functions (such as {@link Patterns#WEB_URL} treat addresses
+     * that omit a protocol (i.e. beginning with 'www') as valid.  However, if you try to actually use this Uri
+     * (e.g. calling CustomTabsIntent.launchUrl(Context, Uri)), it won't work.
+     * <p>
+     * This method will look for the case where the URL string is missing the protocol, and will prepend it if necessary.
+     *
+     * @param inputUri The String that you want to try parsing as a Uri
+     * @param scheme   The scheme (e.g. 'http://', 'https://', 'ftp://') to prepend to this inputUri if it did not have a scheme
+     * @return A valid, well-formed Uri, or an exception if the inputUri parameter was not a valid URI
+     */
+    @NonNull
+    public static Uri parseUriFromString(@NonNull String inputUri, @NonNull String scheme) throws NullPointerException, URISyntaxException {
```

Comment:
I think the NullPointerException is overkill, personally - I mean imagine how many places you could speckle that all over our code :).  URISyntaxException is plenty IMO, esp with the non-null annotations.  *shrug*